owncloudComLibrary/src/test/java/com/owncloud/android/lib/resources/shares/responses/ShareeResponseTest.kt

```diff
@@ -0,0 +1,128 @@
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import java.io.File
+import java.lang.reflect.Type
+
+class ShareeResponseTest {
+
+    lateinit var response: CommonOcsResponse<ShareeOcsResponse>
+
+    private fun loadResponses(fileName: String, adapter: JsonAdapter<CommonOcsResponse<ShareeOcsResponse>>) =
+        adapter.fromJson(File(fileName).readText())
+
+    @Before
+    fun prepare() {
+        val moshi = Moshi.Builder().build()
+        val type: Type = Types.newParameterizedType(CommonOcsResponse::class.java, ShareeOcsResponse::class.java)
+        val adapter: JsonAdapter<CommonOcsResponse<ShareeOcsResponse>> = moshi.adapter(type)
+        response = loadResponses(EXAMPLE_RESPONSE_JSON, adapter)!!
+    }
+
+    @Test
+    fun `check structure - ok - example response files exist`() {
+        val file = File(EXAMPLE_RESPONSE_JSON)
+        assertTrue(file.exists())
+    }
+
+    @Test
+    fun `check structure - ok - contains meta`() {
+        assertEquals("OK", response.ocs.meta.message!!)
+        assertEquals(200, response.ocs.meta.statusCode!!)
+        assertEquals("ok", response.ocs.meta.status!!)
+        assertTrue(response.ocs.meta.itemsPerPage?.isEmpty()!!)
+        assertTrue(response.ocs.meta.totalItems?.isEmpty()!!)
+    }
+
+    @Test
+    fun `check structure - ok - contains exact`() {
+        assertNotEquals(null, response.ocs.data.exact)
+    }
+
+    @Test
+    fun `check structure - ok - contains groups`() {
+        assertNotEquals(null, response.ocs.data.groups)
+    }
+
+    @Test
+    fun `check structure - ok - contains remotes`() {
+        assertNotEquals(null, response.ocs.data.remotes)
+    }
+
+    @Test
+    fun `check structure - ok - contains users`() {
+        assertNotEquals(null, response.ocs.data.users)
+    }
+
+    @Test
+    fun `check structure - ok - groups contains two items`() {
+        assertEquals(2, response.ocs.data.groups.size)
+    }
+
+    @Test
+    fun `check structure - ok - users contains two items`() {
+        assertEquals(2, response.ocs.data.users.size)
+    }
+
+    @Test
+    fun `check structure - ok - exact_users contains one item`() {
+        assertEquals(1, response.ocs.data.exact?.users?.size)
+    }
+
+    @Test
+    fun `check structure - ok - user1 contains additional data`() {
+        assertEquals("user1@user1.com", response.ocs.data.users.get(0).value.additionalInfo)
+    }
+
+    @Test
+    fun `check structure - ok - user2 does not contain additional data`() {
+        assertEquals(null, response.ocs.data.users[1].value.additionalInfo)
+    }
+
+    @Test
+    fun `check empty response - ok - parsing ok`() {
+        val moshi = Moshi.Builder().build()
+        val type: Type = Types.newParameterizedType(CommonOcsResponse::class.java, ShareeOcsResponse::class.java)
+        val adapter: JsonAdapter<CommonOcsResponse<ShareeOcsResponse>> = moshi.adapter(type)
+        loadResponses(EMPTY_RESPONSE_JSON, adapter)!!
+    }
```

Comment:
To test this, you have a `Before` where you load a EXAMPLE_RESPONSE_JSON and then you load an empty one. What are you testing here? I cannot see any assert 🤔